downstairs/src/lib.rs

```diff
@@ -3255,23 +3291,42 @@ pub async fn start_downstairs(
     let dss = d.clone();
     let repair_log = d.lock().await.log.new(o!("task" => "repair".to_string()));
 
-    let repair_listener =
-        match repair::repair_main(&dss, repair_address, &repair_log).await {
-            Err(e) => {
-                // TODO tear down other things if repair server can't be
-                // started?
-                bail!("got {:?} from repair main", e);
-            }
+    let (repair_request_tx, repair_request_rx) = mpsc::channel(10);
+    let repair_listener = match repair::repair_main(
```

Comment:
Instead of spawning a separate task, could we handle this new functionality in `repair::repair_main`?  We're already passing in an `Arc<Mutex<Downstairs>>` (as the `&dss` argument), so I'm not sure what the extra task buys us.

Reply:
We do pass in the downstairs mutex, but we don't keep it.  We only use it to pull out a few fields.
I was actually moddeling this after what you did in `upstairs/src/control.rs` for the single task
refactoring (#1058).

Reply:
I think it's 6 of one, half-dozen of the other.

Having _any task_ holding on to an `Arc<Mutex<Downstairs>>` is something that will have to change if we switch to a single-task implementation (with a dedicated task owning a `Downstairs`).  It doesn't much matter whether the `Arc<Mutex<Downstairs>>` is held by `repair_main` or the new `repair_info`.

Reply:
My hope here was to make an eventual transition to a similar one-task implementation easier, and make the
downstairs similar to the upstairs in that regard.  Is this at least a step in that direction?

Reply:
Previously, there were only two categories of tasks using the `Arc<Mutex<Downstairs>>`:
- The main listener task
- Any tasks spawned by it via `proc_stream`, i.e. `pf_task` and `dw_task`

After this PR, the important change is that we'll have a **third category of task** using the `Arc<Mutex<Downstairs>>`.

This will either be `repair_info` (in the current implementation) or `repair_main` (in my suggestion).  Which one isn't important, the important part (architecturally) is that it's a **third thing** sharing that mutex; and that's what we'll have to deal with in a future single-task refactoring.

From that perspective, either implementation is making a single-task refactoring harder.

That being said, it's not going to make such a refactoring **significantly** harder; we can just make requests to the `Downstairs` task through a queue, instead of calling into its APIs directly.

Since both implementations are architecturally equivalent (w.r.t. a single-task refactoring), I'd vote for the one that doesn't proliferate tasks further (i.e. adding this API to `repair_main`).  Does that make sense?  I don't feel _too_ strongly about it, so I'd defer to your judgement if you still think a separate task is the Right Choice.

Reply:
I put it all in `repair_main()`,  no more repair task.